lib/controllers/login.js

```diff
@@ -33,10 +33,24 @@ module.exports = function(req, res) {
       options = {};
     }
 
+    var hasSocialProviders = false;
+
+    // Iterate all social providers and check if there are any enabled.
+    if (config.socialProviders) {
+      for (var name in config.socialProviders) {
+        var provider = config.socialProviders[name];
+        if (provider.enabled) {
+          hasSocialProviders = true;
+          break;
+        }
+      }
+    }
+
```

Comment:
Yes, for an array. Which this is not.

Also, I thought we already had `underscore` included, but just checked and we don't. So I'll be adding `lodash` instead. I don't think including it is overkill in any sense, because in this case it increases readability and it's very likely that it will be able to improve a lot of the other code too. Though, if it was a module that only did one thing, then I'd definitely agree with you. But in this case I'd say it adds a lot of value to have included in the project.

Also, in the `stormpath` module we're currently using both `underscore` and `lodash`.